### ISSUE TITLE
S2857 triggers on non-SQL string

### ISSUE THREAD

**Issue by hannahchan** [SonarSource/sonar-dotnet#6355]
I have the following code;

```C#
namespace Scaffold.Repositories.Configurations;

using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Scaffold.Domain.Aggregates.Bucket;

public class ItemConfiguration : IEntityTypeConfiguration<Item>
{
    public void Configure(EntityTypeBuilder<Item> builder)
    {
        builder.Property<int>($"{nameof(Bucket)}Id");

        builder.Property(item => item.Id)
            .ValueGeneratedOnAdd();

        builder.HasKey($"{nameof(Bucket)}Id", nameof(Item.Id));

        builder.ToTable(nameof(Item));
    }
}
```

S2857 triggers on the lines;

- `builder.Property<int>($"{nameof(Bucket)}Id");` and
- `builder.HasKey($"{nameof(Bucket)}Id", nameof(Item.Id));`

with the message `Add a space before 'Id'`.

I'm trying to name a column here adding a space is not valid.

I'm using `SonarAnalyzer.CSharp` version `8.48.0.56517`.

**Comment by cristian-ambrosini-sonarsource** [comment]
Thank you for reporting this.
I was able to reproduce the behavior in https://github.com/SonarSource/sonar-dotnet/pull/6614. I labeled the issue as False Positive and it was added to our backlog. Thanks for reporting this, your feedback helps us improve our products.

**Comment by cristian-ambrosini-sonarsource** [comment]
We decided to open a new issue #6633 since, thanks to your feedback, we found that the rule is not raising both for interpolated string with `nameof` expression and interpolated constant string. I will close this issue, please follow the new one to be notified of the fix release. Thanks again for your contribution.
